### PR TITLE
feat: use odpf stencil

### PR DESCRIPTION
**Pull request by sumitaich1998**
refractored dependency package com.gojek.de.stencil to io.odpf.stencil

### CODE REVIEW

**Comment by sumitaich1998** [comment]
When i upgrade Stencil dependency to 0.1.3 , then build fails
When i downgrade Stencil dependency to 0.1.2, build checks pass
cause - Stencil client 0.1.3 uses - group: 'com.datadoghq', name: 'java-dogstatsd-client'
but -Firehose and Stencil client 0.1.2 uses -group: 'com.timgroup', name: 'java-statsd-client'


Reference - [https://github.com/odpf/stencil/commit/0487223034a2ed03b86b9bf94eca90791f498fc1#diff-30fa9b9ffef12b492fe905a950b29ffa748e93e0e0797e91e106d23a9ce44ab2]

File: src/main/java/io/odpf/firehose/config/AppConfig.java [hunk 1]
```diff
@@ -39,6 +39,29 @@
...
+    Long getSchemaRegistryStencilFetchBackoffMinMs();
+
+    @Key("SCHEMA_REGISTRY_STENCIL_FETCH_AUTH_BEARER_TOKEN")
+    String getSchemaRegistryStencilFetchAuthBearerToken();
+
+    @Key("SCHEMA_REGISTRY_STENCIL_CACHE_AUTO_REFRESH")
+    @DefaultValue("false")
+    Boolean getSchemaRegistryStencilCacheAutoRefresh();
+
+    @Key("SCHEMA_REGISTRY_STENCIL_CACHE_TTL_MS")
```
Comment (gauravsinghania) [inline src/main/java/io/odpf/firehose/config/AppConfig.java:61]: TTL_Cache as 15 mins

File: src/main/java/io/odpf/firehose/config/AppConfig.java [hunk 2]
```diff
@@ -39,6 +39,29 @@
...
+    @Key("SCHEMA_REGISTRY_STENCIL_FETCH_TIMEOUT_MS")
+    @DefaultValue("10000")
+    Integer getSchemaRegistryStencilFetchTimeoutMs();
+
+    @Key("SCHEMA_REGISTRY_STENCIL_FETCH_RETRIES")
+    @DefaultValue("4")
+    Integer getSchemaRegistryStencilFetchRetries();
+
+    @Key("SCHEMA_REGISTRY_STENCIL_FETCH_BACKOFF_MIN_MS")
+    @DefaultValue("0L")
```
[the remark this answers is not in the document]
Comment (sumitaich1998) [reply src/main/java/io/odpf/firehose/config/AppConfig.java:51]: fixed

File: src/main/java/io/odpf/firehose/config/AppConfig.java (hunk 1 above)
Comment (sumitaich1998) [reply src/main/java/io/odpf/firehose/config/AppConfig.java:61]: fixed

**Comment by sumitaich1998** [comment]
upgraded stencil to 0.1.3 and replace dependency group: 'com.timgroup','java-statsd-client' with  'com.datadoghq' 'java-dogstatsd-client'

File: docs/assets/telegraf.conf [hunk 3]
```diff
@@ -0,0 +1,994 @@
...
+
+  ## Parses tags in the datadog statsd format
+  ## http://docs.datadoghq.com/guides/dogstatsd/
+  #parse_data_dog_tags = true
+
+  ## Parses datadog extensions to the statsd format
+  datadog_extensions = true
+  ## Parses distributions metric as specified in the datadog statsd format
+  ## https://docs.datadoghq.com/developers/metrics/types/?tab=distribution#definition
+  datadog_distributions = true
```
Comment (sumitaich1998) [inline docs/assets/telegraf.conf:975]: when deployed to production, just check these 2 configs in INPUT PLUGINS, set `datadog_extensions = true` and   `datadog_distributions = true` cc: @jesrypandawa

File: src/main/java/io/odpf/firehose/factory/FactoryUtil.java [hunk 4]
```diff
@@ -60,4 +61,16 @@ public static void configureSubscription(KafkaConsumerConfig config, KafkaConsum
         consumerConfigurationMap.putAll(extraParameters);
         return consumerConfigurationMap;
     }
+
+    public static StencilConfig getStencilConfig(KafkaConsumerConfig kafkaConsumerConfig) {
+        return StencilConfig.builder()
+                .cacheAutoRefresh(kafkaConsumerConfig.getSchemaRegistryStencilCacheAutoRefresh())
```
Comment (prakharmathur82) [inline src/main/java/io/odpf/firehose/factory/FactoryUtil.java:67]: these should come from appConfig rather than kafkaConsumerConfig

**Pull request merged**